### DISCUSSION TITLE
Поддержка Quasar Framework

### DISCUSSION THREAD

**Discussion by aleksei-deets** [yandex-maps-unofficial/vue-yandex-maps#376] [Идеи]
Добый день.
Подскажите пожалуйста, есть ли у бибилиотеки поддержка Vue фреймворка Quasar Framework?

P.S. Было бы здорово если бы её добавили 👍

**Comment by amdest** [comment]
> Добый день. Подскажите пожалуйста, есть ли у бибилиотеки поддержка Vue фреймворка Quasar Framework?
> 
> P.S. Было бы здорово если бы её добавили 👍

Первая версия прекрасно работала, нужно было просто добавить инициализацию в `boot`.
Текущая версия тем более должна без проблем встраиваться в проект Quasar, на мой взгляд нет никакого смысла добавлять какую-либо дополнительную поддержку сторонних фреймворков, тем самым усложняя разработку и уводя её в сторону от основного предназначения.

**Reply by aleksei-deets** [reply]
> > Добый день. Подскажите пожалуйста, есть ли у бибилиотеки поддержка Vue фреймворка Quasar Framework?
> > P.S. Было бы здорово если бы её добавили 👍
> 
> Первая версия прекрасно работала, нужно было просто добавить инициализацию в `boot`. Текущая версия тем более должна без проблем встраиваться в проект Quasar, на мой взгляд нет никакого смысла добавлять какую-либо дополнительную поддержку сторонних фреймворков, тем самым усложняя разработку и уводя её в сторону от основного предназначения.

Если у вас есть опыт подключения, можете показать пример кода для boot

Это было бы полезно для всех кто хочет подключить библиотеку к Quasar

**Reply by amdest** [reply]
> 
> Если у вас есть опыт подключения, можете показать пример кода для boot

```javascript
// файл `/src/boot/<filename>.js`
import { boot } from 'quasar/wrappers';
import YmapPlugin from 'vue-yandex-maps/src';

export default boot(({ app }) => {
  const settings = {
    apiKey: 'xxx-xxx-xxx-xxx',
    lang: 'ru_RU',
    coordorder: 'latlong',
    enterprise: false,
    version: '2.1',
  };

  app.use(YmapPlugin, settings);
});
```

Затем в quasar.config.js в секции boot:

```
...
boot: ['filename', ...] ,
...
```
В компоненте — вызов и работа согласно документации (плагина и Яндекс JS API).

Это для предыдущей версии плагина в варианте `next`, выдержки кода из полностью рабочего проекта.
Для решения одной локальной проблемы с отрисовкой пересекающихся полигонов в проектах использован локальный форк плагина с некоторыми корректировками (поэтому `import YmapPlugin from 'vue-yandex-maps/src';`), но на принцип подключения к приложению они никак не влияли.

В текущей версии плагина не думаю, что принцип включения в проект на Quasar будет иным. С небольшими нюансами, — да. Но кардинально ничего сложного быть не должно.

Я планирую перевести свои проекты с предыдущей версии плагина на текущую, но это сейчас а) не в приоритете и б) много кода придётся переписывать, поэтому сроков нет. Сделаю — дам знать, если вопрос ещё будет актуальным.
